arrow-fx-coroutines/README.MD

```diff
@@ -315,13 +315,82 @@ suspend main(): Unit {
 }
 ```
 
-### Error Handling
+## Concurrency Helpers
+
+### Ref vs Atomic
+
+`Ref` has been renamed `Atomic` in the new API; and, it provides the same level of service as `Ref` 
+(i.e. `Atomic` provides a safe concurrent API to access to a mutable reference).
+
+For example
+
+```kotlin
+fun factorial(n: Int) = IO.fx {
+    val res = Ref(1L).bind()
+
+    (1 until n+1).parTraverse {
+        res.update(it::times)
+    }.bind()
+    res.get().bind()
+}
+```
+
+becomes
+ 
+```kotlin:ank
+ suspend fun factorial(n: Int)  {
+    val res = Atomic(1L)
+
+    (1 until n+1).parTraverse {
+        res.update(it::times)
+    }
+    res.get()
+}```
+
+This code snippet isn’t very useful, except to show how `Ref/Atomic` can be used with concurrent acceses. 
+
+### MVar vs ConcurrentVar
+
+`MVar` is now called `ConcurrentVar`; and, it provides the same level of service as `MVar` 
+(i.e. a `ConcurrentVar` is a mutable concurrent safe variable which is either `empty` or contains a typed `single value`).
+
+```kotlin:ank
+fun sum(state: MVar<ForIO, Int>, list: List<Int>) : IO<Int> =
+    IO.fx {
+        when {
+            list.isEmpty() -> state.take().bind()
+            else -> state.take().flatMap { cur ->
+                state.put(cur + list[0]).flatMap { _ -> sum(state, list.tail()) }
+            }.bind()
+        }
+    }
+```
+
+becomes
+ 
+```kotlin:ank
+suspend fun sum(state: ConcurrentVar<Int>, list: List<Int>) {
```

Comment:
This function can be `tailrec` and then it will be stacksafe ;)

Reply:
Yes thx